src/plugins/data/public/search/es_search/es_search_strategy.ts

```diff
@@ -20,18 +20,25 @@
 import { Observable } from 'rxjs';
 import { ES_SEARCH_STRATEGY, IEsSearchResponse } from '../../../common/search';
 import { SYNC_SEARCH_STRATEGY } from '../sync_search_strategy';
+import { getEsPreference } from './get_es_preference';
 import { TSearchStrategyProvider, ISearchStrategy, ISearchGeneric, ISearchContext } from '..';
 
 export const esSearchStrategyProvider: TSearchStrategyProvider<typeof ES_SEARCH_STRATEGY> = (
   context: ISearchContext,
   search: ISearchGeneric
 ): ISearchStrategy<typeof ES_SEARCH_STRATEGY> => {
   return {
-    search: (request, options) =>
-      search(
+    search: (request, options) => {
+      if (typeof request.params.preference === 'undefined') {
```

Comment:
Would `if (request.params.preference === undefined) {` work?  a bit shorter.    🤷‍♀

Reply:
It's mostly out of habit because you can technically redefine `undefined`:

```
(undefined => {
  console.log(undefined); // 'foo'
})('foo')
```

But yeah, it shouldn't make any difference in this situation.